controllers/gce/utils/utils_test.go

```diff
@@ -0,0 +1,452 @@
+		{"end_with_cluster_name", "560b5b8154db29b3", "default/k8s-tps-e2e-ingress-deployment--560b5b8154db29b3", "default-k8s-tps-e2e-ingress-deployment--560b5b8154db29b3"},
+		{"not_end_with_cluster_name", "560b5b8154db29b3", "default/k8s-tps-e2e-ingress-deployment", "default-k8s-tps-e2e-ingress-deployment--560b5b8154db29b3"},
+		{"will_be_turncted", "560b5b8154db29b3", "default/k8s-tps-e2e-ingress-deployment-xytps-auth", "default-k8s-tps-e2e-ingress-deployment-xytps-auth--560b5b8154d0"},
+	}
+
+	for _, fooTest := range fooTests {
+		t.Run(fooTest.tName, func(t *testing.T) {
+			namer := NewNamer(fooTest.cn, "fooFN")
+			rlbn := namer.LBName(fooTest.k)
+			if rlbn != fooTest.elbn {
+				t.Errorf("Returned %v but expected %v", rlbn, fooTest.elbn)
+			}
+		})
+	}
+}
+
+func TestGetDefaultBackend(t *testing.T) {
+	var nilBS *compute.BackendService
+	bs := &compute.BackendService{CreationTimestamp: "2017-03-08"}
+
+	fooTests := []struct {
+		tName string
+		gm    GCEURLMap
+		ebs   *compute.BackendService
+	}{
+		{"empty_GCEURLMap", map[string]map[string]*compute.BackendService{}, nilBS},
+		{"include_default_but_empty", map[string]map[string]*compute.BackendService{DefaultBackendKey: {}}, nilBS},
+		{"inluce_other_keys_for_default_backend", map[string]map[string]*compute.BackendService{DefaultBackendKey: {DefaultBackendKey + "_non": bs}}, nilBS},
+		{"defalut_backend_has_been_exist", map[string]map[string]*compute.BackendService{DefaultBackendKey: {DefaultBackendKey: bs}}, bs},
+	}
+
+	for _, fooTest := range fooTests {
+		t.Run(fooTest.tName, func(t *testing.T) {
+			rbs := fooTest.gm.GetDefaultBackend()
+			if rbs != fooTest.ebs {
+				t.Errorf("Returned %v but expected %v", rbs, fooTest.ebs)
+			}
+		})
+	}
+}
+
+func TestString(t *testing.T) {
+	fooTests := []struct {
+		tName string
+		gm    GCEURLMap
+		es    string
+	}{
+		{"empty_GCEURLMap", map[string]map[string]*compute.BackendService{}, ""},
+		{"host_exist_url_not_exist", map[string]map[string]*compute.BackendService{"k8s-50xyoeu02x": {}}, "k8s-50xyoeu02x\n"},
+		{"backend_not_exist", map[string]map[string]*compute.BackendService{"k8s-50xyoeu02x": {"http://10.0.0.1/lb": nil}}, "k8s-50xyoeu02x\n\thttp://10.0.0.1/lb: No backend\n"},
+		{"single_host", map[string]map[string]*compute.BackendService{"k8s-50xyoeu02x": {"http://10.0.0.1/lb": {Name: "foo"}}}, "k8s-50xyoeu02x\n\thttp://10.0.0.1/lb: foo\n"},
+		{"multi_host", map[string]map[string]*compute.BackendService{"k8s-50xyoeu02x": {"http://10.0.0.1/lb": {Name: "foo1"}}, "k8s-8xiy02zis": {"http://10.0.0.2/lb": {Name: "foo2"}}}, "k8s-50xyoeu02x\n\thttp://10.0.0.1/lb: foo1\nk8s-8xiy02zis\n\thttp://10.0.0.2/lb: foo2\n"},
+	}
+
+	for _, fooTest := range fooTests {
+		t.Run(fooTest.tName, func(t *testing.T) {
+			rs := fooTest.gm.String()
+			if rs != fooTest.es {
+				t.Errorf("Returned\n%v\nbut expected\n%v", rs, fooTest.es)
+			}
+		})
+	}
+}
+
+func TestPutDefaultBackend(t *testing.T) {
```

Comment:
This is just testing map assignment. The loadbalancer package already has unit tests that assert setting the default, so this is an unnecessary 29 lines.